bpython/history.py

```diff
@@ -232,3 +233,74 @@ def append_reload_and_write(self, s, filename, encoding):
                 # Make sure that entries contains at least one element. If the
                 # file and s are empty, this can occur.
                 self.entries = ['']
+
+class BPythonLinecache(dict):
```

Comment:
I think I'd prefer this in a separate file. `bpython/history.py` is for handling the history file.